File: src/main/scala/com/vc/hard/L128.scala
package com.vc.hard

object L128 {
  def longestConsecutive(arr: Array[Int]): Int = {
    import scala.collection.mutable
    val map = new mutable.HashMap[Int, Int]()
    var res = 0
    arr.indices.foreach(i => {
      if(!map.contains(arr(i))) {
        val left = map.getOrElse(arr(i) - 1, 0)
        val right = map.getOrElse(arr(i) + 1, 0)

        val total = left + 1 + right
        res = Math.max(res, total)
        map.put(arr(i), total)
        map.put(arr(i) - left, total)
        map.put(arr(i) + right, total)
      }
    })
    res
  }
}

File: src/main/java/com/vc/hard/L128.java
package com.vc.hard;

import java.util.*;

class L128 {
    public int longestConsecutive(int[] arr) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int res = 0;
        for(int i = 0; i < arr.length; i++) {
            if(!map.containsKey(arr[i])) {
                int left = map.getOrDefault(arr[i] - 1, 0);
                int right = map.getOrDefault(arr[i] + 1, 0);

                int total = left + right + 1;
                map.put(arr[i], total);
                map.put(arr[i] - left, total);
                map.put(arr[i] + right, total);

                res = Math.max(res, total);
            }
        }
        return res;
    }
}
